src/lib/error_logger.py

```diff
@@ -52,14 +55,59 @@ class ErrorLogger:
     Simple class to be inherited by other classes to add error logging functions.
     """
 
-    logging.basicConfig(format="%(message)s")
+    name: str
+    """ Name of the logger, defaults to the class name. """
+
+    logger: logging.Logger
+    """ Instance of logger which will be used. Each ErrorLogger instance has its own Logger. """
+
+    def __init__(self, name: str = None):
+
+        # Default to the classname
+        self.name = name or self.__class__.__name__
+
+        # Create an instance of logger
+        self.logger = logging.getLogger(self.name)
+
+        # Read logging level from env variable, default to INFO
+        logging_level = {
+            "DEBUG": logging.DEBUG,
+            "INFO": logging.INFO,
+            "WARNING": logging.WARNING,
+            "ERROR": logging.ERROR,
+        }.get(os.getenv("LOG_LEVEL"), logging.INFO)
+        self.logger.setLevel(logging_level)
+
+        # Configure the handler to use our preferred logging format
+        handler = logging.StreamHandler()
+        handler.setFormatter(logging.Formatter("%(message)s"))
+        self.logger.addHandler(handler)
 
     def timestamp(self) -> str:
         return datetime.datetime.now().isoformat()[:24]
 
-    def errlog(self, msg: str, **kwargs) -> None:
-        logging.warning(
+    def _log_msg(self, log_func: Callable, msg: str, **kwargs) -> None:
+        log_func(
             StructuredMessage(
-                msg, timestamp=self.timestamp(), classname=self.__class__.__name__, **kwargs
+                msg,
+                logname=self.name,
+                timestamp=self.timestamp(),
+                # TODO: consider whether we should keep classname or if logname is sufficient
```

Comment:
It seems like we should be fine with logname, only reason to keep it would be to save some work on the Cloud side changing the dashboards and alerting (I currently look for classname). But it wouldn't take too long.

Reply:
Good to know. Then let's keep `classname` for now until the necessary changes are made on the Cloud side.